pkg/aspect/metricsManager.go

```diff
@@ -55,7 +55,7 @@ func newMetricsManager() ReportManager {
 	return &metricsManager{}
 }
 
-func (m *metricsManager) NewReportExecutor(c *cpb.Combined, a adapter.Builder, env adapter.Env, df descriptor.Finder) (ReportExecutor, error) {
+func (m *metricsManager) NewReportExecutor(c *cpb.Combined, createAdapater CreateAspectFunc, env adapter.Env, df descriptor.Finder) (ReportExecutor, error) {
```

Comment:
Adapters implement Aspects, Executors execute Aspects (i.e. Executors care about behaviors, not implementations). So I think CreateAspectFunc is appropriate here.